### ISSUE TITLE
Native compilation of the zig compiler on arm32 fails

### ISSUE THREAD

**Issue by Kuratius** [ziglang/zig#15765]
### Zig Version

commit 7cf2cbb

### Steps to Reproduce and Observed Behavior

Compiled llvm from source using make instead of Ninja using the instructions on the wiki.
https://llvm.org/docs/HowToBuildOnARM.html + the zig wiki.
Used CC=gcc and CXX=g++, on a raspberry pi 4GB running ubuntu server 23.04 32 bit (ARM, not aarch64)
Compiling llvm succeeded.

in the build-release directory (mkdir) in the llvm repo:
CC=gcc CXX=g++ cmake ../llvm   -DCMAKE_INSTALL_PREFIX=$HOME/local/llvm16-release   -DCMAKE_BUILD_TYPE=Release   -DLLVM_ENABLE_PROJECTS="lld;clang"   -DLLVM_ENABLE_LIBXML2=OFF   -DLLVM_ENABLE_TERMINFO=OFF   -DLLVM_ENABLE_LIBEDIT=OFF   -DLLVM_ENABLE_ASSERTIONS=ON   -DLLVM_PARALLEL_LINK_JOBS=1 
make -j3
then in the zig repo in the build directory (mkdir):
 CC=gcc CXX=g++ cmake .. -DCMAKE_PREFIX_PATH=$HOME/local/llvm16-release
make install

Trying to compile zig results in the following error:
[ 65%] Building C object CMakeFiles/zig1.dir/stage1/wasi.c.o
[ 70%] Linking C executable zig1
[ 70%] Built target zig1
[ 75%] Running zig1.wasm to produce /home/ubuntu/zig/build/zig2.c
error: UnknownArchitecture
make[2]: *** [CMakeFiles/zig2.dir/build.make:500: zig2.c] Error 1
make[1]: *** [CMakeFiles/Makefile2:169: CMakeFiles/zig2.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

Is the arm (32 bit) architecture currently not supported?
The same steps produce a working zig compiler on ubuntu server 64 bit.

### Expected Behavior

I expected compilation to succeed.

**Comment by Kuratius** [comment]
It seems that zig tries to build for armv7-linux but the zig compiler doesnt have this target because knows it as arm-linux

make -n outputs this if ran after the make command failed (it does not show this if you run make -n as a dry run, probably because zig's build process is nonstandard)
```/home/ubuntu/zig/build/zig1 /home/ubuntu/zig/lib build-exe src/main.zig -ofmt=c -lc -OReleaseSmall --name zig2 -femit-bin="/home/ubuntu/zig/build/zig2.c" --mod build_options::/home/ubuntu/zig/build/config.zig --deps build_options -target armv7l-linux```

**Comment by Kuratius** [comment]
I fixed these manually:
cd /home/ubuntu/zig && /home/ubuntu/zig/build/zig1 /home/ubuntu/zig/lib build-exe src/main.zig -ofmt=c -lc -OReleaseSmall --name zig2 -femit-bin="/home/ubuntu/zig/build/zig2.c" --mod build_options::/home/ubuntu/zig/build/config.zig --deps build_options -target arm-linux  (had armv7-linux)

cd /home/ubuntu/zig && /home/ubuntu/zig/build/zig1 /home/ubuntu/zig/lib build-obj lib/compiler_rt.zig -ofmt=c -OReleaseSmall --name compiler_rt -femit-bin="/home/ubuntu/zig/build/compiler_rt.c" --mod build_options::/home/ubuntu/zig/build/config.zig --deps build_options -target arm-linux (had armv7-linux)

cd /home/ubuntu/zig && /home/ubuntu/zig/build/zig1 /home/ubuntu/zig/lib build-exe src/main.zig -ofmt=c -lc -OReleaseSmall --name zig2 -femit-bin="/home/ubuntu/zig/build/zig2.c" --mod build_options::/home/ubuntu/zig/build/config.zig --deps build_options -target arm-linux  (had armv7-linux)

cd /home/ubuntu/zig && /home/ubuntu/zig/build/zig1 /home/ubuntu/zig/lib build-obj lib/compiler_rt.zig -ofmt=c -OReleaseSmall --name compiler_rt -femit-bin="/home/ubuntu/zig/build/compiler_rt.c" --mod build_options::/home/ubuntu/zig/build/config.zig --deps build_options -target arm-linux  (had armv7-linux)


now it fails at this step: 
 /usr/bin/gcc  -I/home/ubuntu/local/llvm16-release/include -I/home/ubuntu/zig -I/home/ubuntu/zig/build -I/home/ubuntu/zig/src -I/home/ubuntu/zig/stage1 -g -std=c99 -O0 -fno-stack-protector -MD -MT CMakeFiles/zig2.dir/zig2.c.o -MF CMakeFiles/zig2.dir/zig2.c.o.d -o CMakeFiles/zig2.dir/zig2.c.o -c /home/ubuntu/zig/build/zig2.c
 
 ubuntu@ubuntu:~/zig/build$ /usr/bin/gcc  -I/home/ubuntu/local/llvm16-release/include -I/home/ubuntu/zig -I/home/ubuntu/zig/build -I/home/ubuntu/zig/src -I/home/ubuntu/zig/stage1 -g -std=c99 -O0 -fno-stack-protector -MD -MT CMakeFiles/zig2.dir/zig2.c.o -MF CMakeFiles/zig2.dir/zig2.c.o.d -o CMakeFiles/zig2.dir/zig2.c.o -c /home/ubuntu/zig/build/zig2.c
In file included from /home/ubuntu/zig/build/zig2.c:2:
/home/ubuntu/zig/stage1/zig.h:3193:23: error: initializer element is not constant
 3193 | #define zig_cast_f128 (zig_f128)
      |                       ^
/home/ubuntu/zig/build/zig2.c:1904910: note: in expansion of macro ‘zig_cast_f128’
1904910 | static zig_f128 const math_nan_f128__731 = zig_cast_f128 zig_init_special_f128(, nan, "0x7fff0000000000000000000000000001", zig_init_u128(UINT64_C(0x7fff000000000000), UINT64_C(0x1)));
        |
/home/ubuntu/zig/build/zig2.c: In function ‘fmt_parse_float_convert_fast_fastPow10__anon_206491__206491’:
/home/ubuntu/zig/stage1/zig.h:3193:23: error: initializer element is not constant
 3193 | #define zig_cast_f128 (zig_f128)
      |                       ^
/home/ubuntu/zig/build/zig2.c:2554259: note: in expansion of macro ‘zig_cast_f128’
2554259 |  static zig_f128 const t2[64] = {zig_cast_f128 zig_make_f128(0x1p0, zig_init_u128(UINT64_C(0x3fff000000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.4p3, zig_init_u128(UINT64_C(0x4002400000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.9p6, zig_init_u128(UINT64_C(0x4005900000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.f4p9, zig_init_u128(UINT64_C(0x4008f40000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.388p13, zig_init_u128(UINT64_C(0x400c388000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.86ap16, zig_init_u128(UINT64_C(0x400f86a000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.e848p19, zig_init_u128(UINT64_C(0x4012e84800000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.312dp23, zig_init_u128(UINT64_C(0x4016312d00000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.7d784p26, zig_init_u128(UINT64_C(0x40197d7840000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.dcd65p29, zig_init_u128(UINT64_C(0x401cdcd650000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.2a05f2p33, zig_init_u128(UINT64_C(0x40202a05f2000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.74876e8p36, zig_init_u128(UINT64_C(0x402374876e800000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.d1a94a2p39, zig_init_u128(UINT64_C(0x4026d1a94a200000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.2309ce54p43, zig_init_u128(UINT64_C(0x402a2309ce540000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.6bcc41e9p46, zig_init_u128(UINT64_C(0x402d6bcc41e90000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.c6bf52634p49, zig_init_u128(UINT64_C(0x4030c6bf52634000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.1c37937e08p53, zig_init_u128(UINT64_C(0x40341c37937e0800), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.6345785d8ap56, zig_init_u128(UINT64_C(0x40376345785d8a00), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.bc16d674ec8p59, zig_init_u128(UINT64_C(0x403abc16d674ec80), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.158e460913dp63, zig_init_u128(UINT64_C(0x403e158e460913d0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.5af1d78b58c4p66, zig_init_u128(UINT64_C(0x40415af1d78b58c4), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.b1ae4d6e2ef5p69, zig_init_u128(UINT64_C(0x4044b1ae4d6e2ef5), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.0f0cf064dd592p73, zig_init_u128(UINT64_C(0x40480f0cf064dd59), UINT64_C(0x2000000000000000))),zig_cast_f128 zig_make_f128(0x1.52d02c7e14af68p76, zig_init_u128(UINT64_C(0x404b52d02c7e14af), UINT64_C(0x6800000000000000))),zig_cast_f128 zig_make_f128(0x1.a784379d99db42p79, zig_init_u128(UINT64_C(0x404ea784379d99db), UINT64_C(0x4200000000000000))),zig_cast_f128 zig_make_f128(0x1.08b2a2c28029094p83, zig_init_u128(UINT64_C(0x405208b2a2c28029), UINT64_C(0x940000000000000))),zig_cast_f128 zig_make_f128(0x1.4adf4b7320334b9p86, zig_init_u128(UINT64_C(0x40554adf4b732033), UINT64_C(0x4b90000000000000))),zig_cast_f128 zig_make_f128(0x1.9d971e4fe8401e74p89, zig_init_u128(UINT64_C(0x40589d971e4fe840), UINT64_C(0x1e74000000000000))),zig_cast_f128 zig_make_f128(0x1.027e72f1f12813088p93, zig_init_u128(UINT64_C(0x405c027e72f1f128), UINT64_C(0x1308800000000000))),zig_cast_f128 zig_make_f128(0x1.431e0fae6d7217caap96, zig_init_u128(UINT64_C(0x405f431e0fae6d72), UINT64_C(0x17caa00000000000))),zig_cast_f128 zig_make_f128(0x1.93e5939a08ce9dbd48p99, zig_init_u128(UINT64_C(0x406293e5939a08ce), UINT64_C(0x9dbd480000000000))),zig_cast_f128 zig_make_f128(0x1.f8def8808b02452c9ap102, zig_init_u128(UINT64_C(0x4065f8def8808b02), UINT64_C(0x452c9a0000000000))),zig_cast_f128 zig_make_f128(0x1.3b8b5b5056e16b3be04p106, zig_init_u128(UINT64_C(0x40693b8b5b5056e1), UINT64_C(0x6b3be04000000000))),zig_cast_f128 zig_make_f128(0x1.8a6e32246c99c60ad85p109, zig_init_u128(UINT64_C(0x406c8a6e32246c99), UINT64_C(0xc60ad85000000000))),zig_cast_f128 zig_make_f128(0x1.ed09bead87c0378d8e64p112, zig_init_u128(UINT64_C(0x406fed09bead87c0), UINT64_C(0x378d8e6400000000))),zig_cast_f128 zig_make_f128(0x1.3426172c74d822b878fe8p116, zig_init_u128(UINT64_C(0x40733426172c74d8), UINT64_C(0x22b878fe80000000))),zig_cast_f128 zig_make_f128(0x1.812f9cf7920e2b66973e2p119, zig_init_u128(UINT64_C(0x4076812f9cf7920e), UINT64_C(0x2b66973e20000000))),zig_cast_f128 zig_make_f128(0x1.e17b84357691b6403d0da8p122, zig_init_u128(UINT64_C(0x4079e17b84357691), UINT64_C(0xb6403d0da8000000))),zig_cast_f128 zig_make_f128(0x1.2ced32a16a1b11e8262889p126, zig_init_u128(UINT64_C(0x407d2ced32a16a1b), UINT64_C(0x11e8262889000000))),zig_cast_f128 zig_make_f128(0x1.78287f49c4a1d6622fb2ab4p129, zig_init_u128(UINT64_C(0x408078287f49c4a1), UINT64_C(0xd6622fb2ab400000))),zig_cast_f128 zig_make_f128(0x1.d6329f1c35ca4bfabb9f561p132, zig_init_u128(UINT64_C(0x4083d6329f1c35ca), UINT64_C(0x4bfabb9f56100000))),zig_cast_f128 zig_make_f128(0x1.25dfa371a19e6f7cb54395cap136, zig_init_u128(UINT64_C(0x408725dfa371a19e), UINT64_C(0x6f7cb54395ca0000))),zig_cast_f128 zig_make_f128(0x1.6f578c4e0a060b5be2947b3c8p139, zig_init_u128(UINT64_C(0x408a6f578c4e0a06), UINT64_C(0xb5be2947b3c8000))),zig_cast_f128 zig_make_f128(0x1.cb2d6f618c878e32db399a0bap142, zig_init_u128(UINT64_C(0x408dcb2d6f618c87), UINT64_C(0x8e32db399a0ba000))),zig_cast_f128 zig_make_f128(0x1.1efc659cf7d4b8dfc904004744p146, zig_init_u128(UINT64_C(0x40911efc659cf7d4), UINT64_C(0xb8dfc90400474400))),zig_cast_f128 zig_make_f128(0x1.66bb7f0435c9e717bb45005915p149, zig_init_u128(UINT64_C(0x409466bb7f0435c9), UINT64_C(0xe717bb4500591500))),zig_cast_f128 zig_make_f128(0x1.c06a5ec5433c60ddaa16406f5a4p152, zig_init_u128(UINT64_C(0x4097c06a5ec5433c), UINT64_C(0x60ddaa16406f5a40))),zig_cast_f128 zig_make_f128(0x1.18427b3b4a05bc8a8a4de8459868p156, zig_init_u128(UINT64_C(0x409b18427b3b4a05), UINT64_C(0xbc8a8a4de8459868))),zig_cast_f128 zig_make_f128(0x1.5e531a0a1c872bad2ce16256fe82p159, zig_init_u128(UINT64_C(0x409e5e531a0a1c87), UINT64_C(0x2bad2ce16256fe82))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0)))};
        |
/home/ubuntu/zig/stage1/zig.h:3193:23: note: (near initialization for ‘t2[0]’)
 3193 | #define zig_cast_f128 (zig_f128)
      |                       ^
/home/ubuntu/zig/build/zig2.c:2554259: note: in expansion of macro ‘zig_cast_f128’
2554259 |  static zig_f128 const t2[64] = {zig_cast_f128 zig_make_f128(0x1p0, zig_init_u128(UINT64_C(0x3fff000000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.4p3, zig_init_u128(UINT64_C(0x4002400000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.9p6, zig_init_u128(UINT64_C(0x4005900000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.f4p9, zig_init_u128(UINT64_C(0x4008f40000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.388p13, zig_init_u128(UINT64_C(0x400c388000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.86ap16, zig_init_u128(UINT64_C(0x400f86a000000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.e848p19, zig_init_u128(UINT64_C(0x4012e84800000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.312dp23, zig_init_u128(UINT64_C(0x4016312d00000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.7d784p26, zig_init_u128(UINT64_C(0x40197d7840000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.dcd65p29, zig_init_u128(UINT64_C(0x401cdcd650000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.2a05f2p33, zig_init_u128(UINT64_C(0x40202a05f2000000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.74876e8p36, zig_init_u128(UINT64_C(0x402374876e800000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.d1a94a2p39, zig_init_u128(UINT64_C(0x4026d1a94a200000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.2309ce54p43, zig_init_u128(UINT64_C(0x402a2309ce540000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.6bcc41e9p46, zig_init_u128(UINT64_C(0x402d6bcc41e90000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.c6bf52634p49, zig_init_u128(UINT64_C(0x4030c6bf52634000), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.1c37937e08p53, zig_init_u128(UINT64_C(0x40341c37937e0800), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.6345785d8ap56, zig_init_u128(UINT64_C(0x40376345785d8a00), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.bc16d674ec8p59, zig_init_u128(UINT64_C(0x403abc16d674ec80), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.158e460913dp63, zig_init_u128(UINT64_C(0x403e158e460913d0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.5af1d78b58c4p66, zig_init_u128(UINT64_C(0x40415af1d78b58c4), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.b1ae4d6e2ef5p69, zig_init_u128(UINT64_C(0x4044b1ae4d6e2ef5), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x1.0f0cf064dd592p73, zig_init_u128(UINT64_C(0x40480f0cf064dd59), UINT64_C(0x2000000000000000))),zig_cast_f128 zig_make_f128(0x1.52d02c7e14af68p76, zig_init_u128(UINT64_C(0x404b52d02c7e14af), UINT64_C(0x6800000000000000))),zig_cast_f128 zig_make_f128(0x1.a784379d99db42p79, zig_init_u128(UINT64_C(0x404ea784379d99db), UINT64_C(0x4200000000000000))),zig_cast_f128 zig_make_f128(0x1.08b2a2c28029094p83, zig_init_u128(UINT64_C(0x405208b2a2c28029), UINT64_C(0x940000000000000))),zig_cast_f128 zig_make_f128(0x1.4adf4b7320334b9p86, zig_init_u128(UINT64_C(0x40554adf4b732033), UINT64_C(0x4b90000000000000))),zig_cast_f128 zig_make_f128(0x1.9d971e4fe8401e74p89, zig_init_u128(UINT64_C(0x40589d971e4fe840), UINT64_C(0x1e74000000000000))),zig_cast_f128 zig_make_f128(0x1.027e72f1f12813088p93, zig_init_u128(UINT64_C(0x405c027e72f1f128), UINT64_C(0x1308800000000000))),zig_cast_f128 zig_make_f128(0x1.431e0fae6d7217caap96, zig_init_u128(UINT64_C(0x405f431e0fae6d72), UINT64_C(0x17caa00000000000))),zig_cast_f128 zig_make_f128(0x1.93e5939a08ce9dbd48p99, zig_init_u128(UINT64_C(0x406293e5939a08ce), UINT64_C(0x9dbd480000000000))),zig_cast_f128 zig_make_f128(0x1.f8def8808b02452c9ap102, zig_init_u128(UINT64_C(0x4065f8def8808b02), UINT64_C(0x452c9a0000000000))),zig_cast_f128 zig_make_f128(0x1.3b8b5b5056e16b3be04p106, zig_init_u128(UINT64_C(0x40693b8b5b5056e1), UINT64_C(0x6b3be04000000000))),zig_cast_f128 zig_make_f128(0x1.8a6e32246c99c60ad85p109, zig_init_u128(UINT64_C(0x406c8a6e32246c99), UINT64_C(0xc60ad85000000000))),zig_cast_f128 zig_make_f128(0x1.ed09bead87c0378d8e64p112, zig_init_u128(UINT64_C(0x406fed09bead87c0), UINT64_C(0x378d8e6400000000))),zig_cast_f128 zig_make_f128(0x1.3426172c74d822b878fe8p116, zig_init_u128(UINT64_C(0x40733426172c74d8), UINT64_C(0x22b878fe80000000))),zig_cast_f128 zig_make_f128(0x1.812f9cf7920e2b66973e2p119, zig_init_u128(UINT64_C(0x4076812f9cf7920e), UINT64_C(0x2b66973e20000000))),zig_cast_f128 zig_make_f128(0x1.e17b84357691b6403d0da8p122, zig_init_u128(UINT64_C(0x4079e17b84357691), UINT64_C(0xb6403d0da8000000))),zig_cast_f128 zig_make_f128(0x1.2ced32a16a1b11e8262889p126, zig_init_u128(UINT64_C(0x407d2ced32a16a1b), UINT64_C(0x11e8262889000000))),zig_cast_f128 zig_make_f128(0x1.78287f49c4a1d6622fb2ab4p129, zig_init_u128(UINT64_C(0x408078287f49c4a1), UINT64_C(0xd6622fb2ab400000))),zig_cast_f128 zig_make_f128(0x1.d6329f1c35ca4bfabb9f561p132, zig_init_u128(UINT64_C(0x4083d6329f1c35ca), UINT64_C(0x4bfabb9f56100000))),zig_cast_f128 zig_make_f128(0x1.25dfa371a19e6f7cb54395cap136, zig_init_u128(UINT64_C(0x408725dfa371a19e), UINT64_C(0x6f7cb54395ca0000))),zig_cast_f128 zig_make_f128(0x1.6f578c4e0a060b5be2947b3c8p139, zig_init_u128(UINT64_C(0x408a6f578c4e0a06), UINT64_C(0xb5be2947b3c8000))),zig_cast_f128 zig_make_f128(0x1.cb2d6f618c878e32db399a0bap142, zig_init_u128(UINT64_C(0x408dcb2d6f618c87), UINT64_C(0x8e32db399a0ba000))),zig_cast_f128 zig_make_f128(0x1.1efc659cf7d4b8dfc904004744p146, zig_init_u128(UINT64_C(0x40911efc659cf7d4), UINT64_C(0xb8dfc90400474400))),zig_cast_f128 zig_make_f128(0x1.66bb7f0435c9e717bb45005915p149, zig_init_u128(UINT64_C(0x409466bb7f0435c9), UINT64_C(0xe717bb4500591500))),zig_cast_f128 zig_make_f128(0x1.c06a5ec5433c60ddaa16406f5a4p152, zig_init_u128(UINT64_C(0x4097c06a5ec5433c), UINT64_C(0x60ddaa16406f5a40))),zig_cast_f128 zig_make_f128(0x1.18427b3b4a05bc8a8a4de8459868p156, zig_init_u128(UINT64_C(0x409b18427b3b4a05), UINT64_C(0xbc8a8a4de8459868))),zig_cast_f128 zig_make_f128(0x1.5e531a0a1c872bad2ce16256fe82p159, zig_init_u128(UINT64_C(0x409e5e531a0a1c87), UINT64_C(0x2bad2ce16256fe82))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0))),zig_cast_f128 zig_make_f128(0x0.0p0, zig_init_u128(UINT64_C(0x0), UINT64_C(0x0)))};
        |
ubuntu@ubuntu:~/zig/build$

**Comment by Kuratius** [comment]
Issue persists with newest commit.

**Comment by andrewrk** [comment]
Fixed by #16246. Tell me if the issue persists and I'll re-open.

**Comment by Kuratius** [comment]
The issue described here was fixed, but there are other steps further in the compilation process that are still failing. 

jacoby suggested trying thumb first, as that was what gcc was defaulting to on my system, but didn't get past a out of memory crash with -Os, without that option it failed at a different compilation step where the error was somewhat vague (related to compilation units being more than 16MB).

I think I was supposed to check if using led gets around the memory crash, but I never got around to it.

I think we gave up on it for the time being.

FreshCoffee said he would take a look at fixing zig-bootstrap, but I don't know if that has been done.
There is still a hardcoded error that should probably  be removed, otherwise that will probably fail too.

https://github.com/ziglang/zig/pull/16246#issuecomment-1620864074
